src/eve/message/file.rs

```diff
@@ -0,0 +1,28 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum FileState {
+    #[serde(rename = "OPEN")]
+    Open,
+    #[serde(rename = "CLOSED")]
+    Closed,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct FileInfo {
```

Comment:
Doesn't this need to account for the potential hash fields?

Also, is there anything need for the "force-magic" config?

Reply:
Actually, I guess these would also need to be exposed here[1]?

[1] https://github.com/protectwise/suricata-ipc/pull/33/files#diff-29edda423cb602876706ea17ca58531b45401f5141999632926eeb8dde628edeR104

Reply:
Wasn't going to add file hash support to this PR. Could do another PR if that is desired. 

Magic is currently defaulted to off. Same thing, could do a follow on PR to log magic.

Reply:
Rgr, we'll definitely need at least one hash of the file but looking at the docs, it seems the file-id used to write to disk is the SHA256. Is that the "sid" in the fileInfo?

Reply:
To actually store them on disk, you need a rule https://github.com/protectwise/suricata-ipc/pull/33/commits/d1516c13d5d2660d8b22931999f1dea2f179318a#diff-10175120aeda1f9b3962be0bcbf25906e6e88636bd37e54fdb04cfaee961e906R23

https://suricata.readthedocs.io/en/suricata-6.0.0/file-extraction/file-extraction.html#rules

I'll add support for hash.

Reply:
https://github.com/protectwise/suricata-ipc/pull/33/commits/6dda6f1924db39acf3825546deade089facf5ad4